packages/notebook/src/actions.tsx

```diff
@@ -1219,6 +1303,12 @@ namespace NotebookActions {
  * A namespace for private data.
  */
 namespace Private {
+  /**
+   * A signal that emits whenever a cell is run.
+   */
+  export
+  const executed = new Signal<any, { parent: Notebook, child: Cell }>({ });
```

Comment:
What winds up being bound to the first argument of the signal template? Also, a couple of thoughts about how this would be used:
1. Would `notebook` and `cell` be better names for the object attributes?
2. What if somebody wanted to know the index of the cell that was executed? They could iterate over `notebook.widgets` to find a match for `cell`, but perhaps we should bundle that information in the args?

Reply:
We don't have that index either at the point where the signal is emitted. I can see it being useful but I am not sure I want to incur the lookup every time a cell is executed, especially because many cells might run in a row and each of them would require a linear search.

I am not wedded to `parent` and `child`. I'm happy to change those to `notebook` and `cell`.

Reply:
True enough, it just puts the onus of the linear search on the consumer of the signal (which they may or may not need to do).

How about the `any` for the first template argument? What gets bound to that if the `NotebookActions` class is never instantiated?

Reply:
The `Signal` instance has an object that it is instantiated with:

```
const executed = new Signal<any, { notebook: Notebook, cell: Cell }>({ });
```

I would have made it `sender: void` or `sender: null` but the underlying `Signal` class implementation uses the senders and receivers in a weak map, so you can't use `undefined` or `null` as keys in that map. A consumer of this signal could reference the `sender` but it'll be harmless and effectively meaningless. We have several signals where we effectively ignore the sender even though it has a type because we often `connect()` in a spot that already has a reference to that sender.

Reply:
`index = notebook.widgets.indexOf(cell)` is simple enough that since it's possible many connectors to this signal won't need it, I don't think it's too bad to impose it on those who do ... although, I suppose the risk of that is if there are ten signal connections and each one actually *does* care about that index, then we've imposed a 10x heavier CPU load than if the index were calculated upfront.

I think that since adding the index would be API-compatible, maybe we should wait to see how it is used. @mkery might have an opinion about this.

Reply:
That's  reasonable. Thanks @afshin!